lib/child_process.js

```diff
@@ -418,13 +464,34 @@ function spawnSync(/*file, args, options*/) {
 
   debug('spawnSync', opts.args, options);
 
+  // Validate the timeout, if present.
+  if (options.timeout != null &&
+      !(typeof options.timeout === 'number' && options.timeout >= 0)) {
+    throw new TypeError('"timeout" must be a non-negative number');
```

Comment:
It has to be an integer. The C++ code does `js_timeout->IntegerValue()`.
Suggestion: use `Number.isInteger(options.timeout) && options.timeout >= 0`